docs/HTTP-batchsource.md

```diff
@@ -0,0 +1,206 @@
+
+**Retry Policy:** Policy used to calculate delay between retries.
+
+**Linear Retry Interval:** Interval between retries. Is only used if retry policy is "linear".
+
+**Max Retry Duration:** Maximum time in seconds retries can take.
+
+**Connect Timeout:** Maximum time in seconds connection initialization is allowed to take.
+
+**Read Timeout:** Maximum time in seconds fetching data from the server is allowed to take.
+
+### Pagination
+
+**Pagination Type:** Strategy used to determine how to get next page.
+
+**Wait time between pages:** Interval between loading pages in milliseconds.
+<br><br>
+
+##### Pagination type: None
+Only single page is loaded.
+<br><br>
+
+##### Pagination type: Link in response header
+In response there is a "Link" header, which contains an url marked as "next". Example:<br>
+`<http://example.cdap.io/admin/api/pages?page=1&q.language.id=1>; rel="first",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="next",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="last"`
+<br><br>
+
+##### Pagination type: Link in response body
+Every page contains a next page url. This pagination type is only supported for JSON and XML formats.
+Pagination happens until no next page field is present or until page contains no elements.
+
+**Next Page JSON/XML Field Path:** A JSON path or an XPath to a field which contains next page url.
+<br><br>
+
+##### Pagination type: Token in response body
+Every page contains a token, which is appended as an url parameter to obtain next page.
+This type of pagination is only supported for JSON and XML formats. Pagination happens until no next page
+token is present on the page or until page contains no elements.
+
+**Next Page Token Path:** A JSON path or an XPath to a field which contains next page token.
+
+**Next Page Url Parameter:** A parameter which is appended to url in order to specify next page token.
+<br><br>
+
+##### Pagination type: Increment an index
+Pagination by incrementing a {pagination.index} placeholder value in url. For this pagination type url is required
+to contain above placeholder.
+
+**Start Index:** Start value of {pagination.index} placeholder
+
+**Max Index:** Maximum value of {pagination.index} placeholder. If empty, pagination will happen until the page with
+no elements.
+
+**Index Increment:** A value which the {pagination.index} placeholder is incremented by. Increment can be negative.
+<br><br>
+
+##### Pagination type: Custom
+Pagination using user provided code. The code decides how to retrieve a next page url based on previous page contents
+and headers and when to finish pagination.
+
+**Custom Pagination Python Code:** A code which implements retrieving
+a next page url based on previous page contents and headers.
+
+### OAuth2
+
+**OAuth2 Enabled:** If true, plugin will perform OAuth2 authentication.
+
+**Auth URL:** The endpoint for authorization server, which retrieves the authorization code.
```

Comment:
By convention, we don't start descriptions with 'The', otherwise most descriptions would start with 'The'.

Endpoint for the authorization server used to retrieve the authorization code.

Reply:
done